### ISSUE TITLE
Upgrading to git 2.8 returns null sha

### ISSUE THREAD

**Issue by ghost** [rwjblue/git-repo-info#16]
``` sh
❯ git --version
git version 2.8.1
```

``` js
❯ nesh
Node v5.5.0
Type .help for more information
nesh> var getRepoInfo = require('git-repo-info');
undefined
nesh> var info = getRepoInfo();
undefined
nesh> info
{ sha: null,
  abbreviatedSha: null,
  branch: 'some-branch',
  tag: null }
```

Other members using this with git versions lower than 2.8 report no problems.

**Comment by rwjblue** [comment]
Thanks for reporting, I don't see anything obvious in https://git.kernel.org/cgit/git/git.git/tree/Documentation/RelNotes/2.8.0.txt?h=v2.8.1 that would point us in the right direction.  

I'm still running 2.4.2 locally (which works properly). I think we need to test to figure out which version introduced the incompatibility.

**Comment by seawatts** [comment]
When I run this locally with 2.7.2 and 2.8.2 it works correctly. However using 2.7.4 on circle.ci it returns null.

**Comment by seawatts** [comment]
Oh I just noticed after sshing into circle ci that the ./git/refs/heads/my-branch-name doesn't exist.

**Comment by seawatts** [comment]
I just talked with someone from circle CI. They said they do a git shallow copy.

**Comment by seawatts** [comment]
I think this is related to #17

**Comment by rwjblue** [comment]
https://github.com/rwjblue/git-repo-info/pull/18 should have fixed, and is released in 1.1.3.
